src/main/java/org/jitsi/videobridge/cc/vp8/VP8AdaptiveTrackProjectionContext.java

```diff
@@ -179,6 +189,52 @@ public VP8AdaptiveTrackProjectionContext(
         return null;
     }
 
+    /** Clean up old entries in the frame projection map.
+     *
+     * The frame projection map is maintained as a sorted map; to clean it up,
+     * we walk entries from the oldest (in timestamp ordering), removing anything
+     * more than #WAIT_SAMPLES older than the newest timestamp.
+     *
+     * Because the map is ordered, we can stop searching entries when we reach
+     * one that is new enough not to be removed, so this is amortized constant
+     * time per frame.
+     *
+     * Caller should be synchronized.
+     * */
+    private
+    void cleanupFrameProjectionMap(long newTimestamp)
+    {
+        if (vp8FrameProjectionMap.isEmpty())
+        {
+            return;
+        }
+
+        long newestTimestamp = vp8FrameProjectionMap.lastKey();
+
+        /* If our timestamps have jumped by a quarter of the number space, reset the map. */
+        long timestampJump = RtpUtils.getTimestampDiff(newTimestamp, newestTimestamp);
+        if (timestampJump >= 0x4000_0000 || timestampJump <= -0x4000_0000)
+        {
+            vp8FrameProjectionMap.clear();
+            return;
+        }
+
+        long threshold = RtpUtils.applyTimestampDelta(newTimestamp, -WAIT_SAMPLES);
+
+        Iterator<Long> it = vp8FrameProjectionMap.keySet().iterator();
+        while (it.hasNext())
+        {
+            Long key = it.next();
+            if (RtpUtils.isNewerTimestampThan(threshold, key))
```

Comment:
i'd expect this to check if it's older than the threshold not newer?

Reply:
It's checking whether the threshold is newer than the key, rather than if the key is older than the threshold - so it's correct but confusing. I'll reverse it.